scanpipe/pipes/d2d.py

```diff
@@ -1343,3 +1347,43 @@ def flag_deployed_from_resources_with_missing_license(project, doc_extensions=No
 
     no_license_files.update(status=flag.NO_LICENSES)
     unknown_license_files.update(status=flag.UNKNOWN_LICENSE)
+
+
+def handle_dangling_deployed_legal_files(project, logger):
+    """
+    Scan the legal files with empty status and update status
+    to `REVIEW_DANGLING_LEGAL_FILE`.
+    """
+    to_resources = project.codebaseresources.files().to_codebase().no_status()
+
+    legal_file_filter = Q()
+
+    for token in LEGAL_STARTS_ENDS:
+        legal_file_filter |= Q(name__istartswith=token)
+        legal_file_filter |= Q(name__iendswith=token)
+        legal_file_filter |= Q(name__iendswith=Concat(Value(token), F("extension")))
```

Comment:
What's a data example of this case?

Reply:
Examples of these are covered in unit test.
https://github.com/nexB/scancode.io/blob/613f9fe0a951515f7adadef0715717ffa2060b70/scanpipe/tests/pipes/test_d2d.py#L1192-L1194

Reply:
I don't see the `Q(name__iendswith=Concat(Value(token), F("extension")))` case.
Those are filtered using `name__iendswith` and `name__istartswith`, no?

Reply:
`project_notice.txt` would not be covered by  `name__iendswith` and `name__istartswith` since the name in itself doesn't starts or ends with token `notice`.

Reply:
Fair enough, your test is missing a case such as `project.LICENSE` to cover the `Q(name__iendswith=token)`.